config/action.d/pf.conf

```diff
@@ -64,9 +79,24 @@ actioncheck = <pfctl> -sr | grep -q <tablename>-<name>
 # Tags:    <ip>  IP address
 #          <failures>  number of failures
 #          <time>  unix timestamp of the ban time
+#          <port>  ports being banned, if using multiport action
 # Values:  CMD
 #
-actionban = <pfctl> -t <tablename>-<name> -T add <ip>
+actionban = <pfctl> -t <tablename>-<name> -T add <ip>; if [ "<kick>" = "true" ]; then
+           RANGE_TEST=`echo "<port>" | sed "s/://"`
+           NUM_TEST=`echo "<port>" | sed "s/[A-Za-z]//"`
+           if [ "<actiontype>" = "any" ]; then
+               pfctl -k <ip>
+           elif [ "$RANGE_TEST" != "<port>" ]; then
```

Comment:
I don't think it is good idea to validate action parameters here (every time for every ban)...
1. Banning actions are not supposed to produce an error (unfortunately we don't have currently a facility to validate parameters).
2. There is `actionstart` which will be executed once

Reply:
Ohh! Yeah. That would definitely be a problem.

So better to fail during action start then.

Fewer context switches as well.